WordPress/src/test/java/org/wordpress/android/ui/reader/discover/interests/ReaderInterestsViewModelTest.kt

```diff
@@ -66,20 +74,134 @@ class ReaderInterestsViewModelTest {
             initViewModel()
 
             // Then
-            assertThat(viewModel.uiState.value).isNull()
+            assertThat(viewModel.uiState.value).isEqualTo(
+                UiState(
+                    listOf(),
+                    ReaderTagList(),
+                    DoneButtonHiddenUiState
+                )
+            )
+        }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun `done button hidden on start switches to disabled state when tags received from repo`() =
+        coroutineScope.runBlockingTest {
+            // Given
+            val mockInterests = getMockInterests()
+            whenever(readerTagRepository.getInterests()).thenReturn(mockInterests)
+
+            // Pause dispatcher so we can verify done button initial state
+            coroutineScope.pauseDispatcher()
+
+            // Trigger data load
+            initViewModel()
+
+            assertThat(requireNotNull(viewModel.uiState.value).doneButtonUiState)
+                .isInstanceOf(DoneButtonHiddenUiState::class.java)
+
+            // Resume pending coroutines execution
+            coroutineScope.resumeDispatcher()
+
+            assertThat(requireNotNull(viewModel.uiState.value).doneButtonUiState)
+                .isInstanceOf(DoneButtonDisabledUiState::class.java)
+        }
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun `done button enabled if at least one interest selected`() =
+        coroutineScope.runBlockingTest {
+            // Given
+            val mockInterests = getMockInterests()
+            whenever(readerTagRepository.getInterests()).thenReturn(mockInterests)
+
+            // When
+            initViewModel()
+            viewModel.onInterestAtIndexToggled(index = 0, isChecked = true)
+
+            // Then
+            assertThat(requireNotNull(viewModel.uiState.value).interestsUiState.filter { it.isChecked }.size)
```

Comment:
I feel this assertion doesn't belong to this test. Each test should ideally have a single purpose/focus/reasonToFail.  I'd consider creating a new test "Interest selected after onInterestAtIndexToggled invoked" or similar. Wdyt?

Reply:
Refactored here: a456645